experimental/gittuf/hook.go

```diff
@@ -22,8 +28,77 @@ func (e *ErrHookExists) Error() string {
 
 type HookType string
 
+var (
+	ErrNoHooksFoundForPrincipal = errors.New("no hooks found for the specified principal")
+)
+
 var HookPrePush = HookType("pre-push")
 
+// InvokeHook runs the hooks defined in the specified stage for the user defined
+// by principalID. Upon successful completion of all hooks for the stage for the
+// user, the map of hook names to exit codes is returned.
+// TODO: Add attestations workflow
+func (r *Repository) InvokeHook(ctx context.Context, stage tuf.HookStage, principalID, targetsRoleName string, parameters ...string) (map[string]int, error) {
```

Comment:
nit: the name of this API indicates that a specific hook will be invoked, when this actually identifies what hook(s) to execute.

Reply:
Have we thought through how the delegations graph is parsed to identify what programs to load and execute?

edit: ah they only live in the root metadata